### PR TITLE
Fix storyshots by moving cacheDirectory to webpack config

### LINKED ISSUE #1709: Unknown option: base.cacheDirectory
I was trying the new `3.2.6-alpha.0` release but it has an error when used with `storyshots` because of https://github.com/storybooks/storybook/blame/master/app/react/src/server/config/babel.js#L6 which was recently added.

We must omit that in https://github.com/storybooks/storybook/blob/master/addons/storyshots/src/index.js#L41 and probably other places.

//cc @ndelangen

### PR DESCRIPTION
**Pull request by ndelangen**
fixes: #1709 

## What I did
Moved cacheDirectory back to webpack config from babel config

## How to test
See issue (I think it should be apparent when using storyshots?)

### CODE REVIEW

**Review by Hypnosphi — APPROVED** [review]
LGTM

**Comment by tmeasday** [comment]
RE: testing -- yes this was a storyshots issue, test the various app modes in SS.

File: app/vue/src/server/config/babel.js [hunk 1]
```diff
@@ -3,7 +3,6 @@ const findCacheDir = require('find-cache-dir');
 module.exports = {
```
Comment (satazor) [inline app/vue/src/server/config/babel.js]: Remove unused findCacheDir above.

File: app/vue/src/server/config.js [hunk 2]
```diff
@@ -11,10 +12,16 @@ const logger = console;
 // (inside working directory) if a config path is not provided.
 export default function(configType, baseConfig, configDir) {
   const config = baseConfig;
-
+
```
Comment (satazor) [inline app/vue/src/server/config.js]: Remove extra space.

File: app/vue/src/server/config.js [hunk 3]
```diff
@@ -11,10 +12,16 @@ const logger = console;
 // (inside working directory) if a config path is not provided.
 export default function(configType, baseConfig, configDir) {
   const config = baseConfig;
-
+  
   const babelConfig = loadBabelConfig(configDir);
-  config.module.rules[0].query = babelConfig;
-
+  config.module.rules[0].query = {
```
Comment (satazor) [inline app/vue/src/server/config.js]: Shouldn't this be applied to the react app too?

**Review by satazor — CHANGES_REQUESTED** [review]

**Comment by Hypnosphi** [comment]
Seems like we `.eslintignore` everything related to vue: https://github.com/storybooks/storybook/blob/master/.eslintignore#L9
Is there any good reason for that?

**Comment by Hypnosphi** [comment]
That's what I get when I remove this line:
```
111 problems (10 errors, 101 warnings)
0 errors, 85 warnings potentially fixable with the `--fix` option.
```

Seems doable, I'll try to handle it

**Comment by Hypnosphi** [comment]
Here it is #1715

**Comment by satazor** [comment]
❤️ 

Could we have a patch release? :D

**Pull request merged**